code/game/mecha/combat/nerchen.dm

```diff
@@ -0,0 +1,211 @@
+		to_chat(user, "<span class='warning'>You are currently buckled and cannot move.</span>")
+		log_message("Permission denied (Buckled).", LOG_MECHA)
+		return
+	if(user.has_buckled_mobs()) //mob attached to us
+		to_chat(user, "<span class='warning'>You can't enter the exosuit with other creatures attached to you!</span>")
+		log_message("Permission denied (Attached mobs).", LOG_MECHA)
+		return
+
+	ner.visible_message("<span class='notice'>[user] starts to climb into [name].</span>")
+
+	if(do_after(user, enter_delay, target = ner)) //chen will be where ner is, so this lets the progressbar be shown
+		if(obj_integrity <= 0)
+			to_chat(user, "<span class='warning'>You cannot get in the [name], it has been destroyed!</span>")
+		else if(occupant)
+			to_chat(user, "<span class='danger'>[occupant] was faster! Try better next time, loser.</span>")
+		else if(user.buckled)
+			to_chat(user, "<span class='warning'>You can't enter the exosuit while buckled.</span>")
+		else if(user.has_buckled_mobs())
+			to_chat(user, "<span class='warning'>You can't enter the exosuit with other creatures attached to you!</span>")
+		else
+			moved_inside(user)
+	else
+		to_chat(user, "<span class='warning'>You stop entering the exosuit!</span>")
+	return
+
+/obj/mecha/combat/chen/moved_inside(mob/living/carbon/human/H)
+	. = ..()
+	ner.cockpit_icon()
+
+/obj/mecha/combat/chen/domove(direction)
+	to_chat(occupant, "<span class='warning'>You need to be in the other cockpit to move!</span>")
+	return FALSE
+
+/obj/mecha/combat/chen/go_out(forced, atom/newloc = ner.loc)
+	..()
+	ner.cockpit_icon()
+
+/obj/mecha/combat/chen/click_action(atom/target,mob/user,params, chencommand = FALSE)
+	ner.click_action(target, user, params, chencommand = TRUE)
+
+#define LEAP_COOLDOWN 30 SECONDS
+
+//action buttons
+/datum/action/innate/mecha/leap
+	name = "Mecha Leap"
+	button_icon_state = "mech_missile"
+	var/cooldowns = 0 //i know we have a cooldown action but other mecha action buttons have not used it
+
+/datum/action/innate/mecha/leap/Activate(forced_state = null)
+	if(!owner || !chassis || chassis.occupant != owner)
+		return
+	if(!isnull(forced_state))
+		chassis.leg_overload_mode = forced_state
+	else
+		chassis.leg_overload_mode = !chassis.leg_overload_mode
+	button_icon_state = "mech_overload_[chassis.leg_overload_mode ? "on" : "off"]"
+	chassis.log_message("Leaped forward.", LOG_MECHA)
+	if(chassis.leg_overload_mode)
+		chassis.leg_overload_mode = 1
+		chassis.step_in = min(1, round(chassis.step_in/2))
+		chassis.step_energy_drain = max(chassis.overload_step_energy_drain_min,chassis.step_energy_drain*chassis.leg_overload_coeff)
+		chassis.occupant_message("<span class='danger'>You enable leg actuators overload.</span>")
+	else
+		chassis.leg_overload_mode = 0
+		chassis.step_in = initial(chassis.step_in)
+		chassis.step_energy_drain = chassis.normal_step_energy_drain
+		chassis.occupant_message("<span class='notice'>You disable leg actuators overload.</span>")
```

Comment:
```suggestion
	chassis.step_in = chassis.leg_overload_mode ? min(1, round(chassis.step_in/2)) : initial(chassis.step_in)
	chassis.step_energy_drain = chassis.leg_overload_mode ? max(chassis.overload_step_energy_drain_min,chassis.step_energy_drain*chassis.leg_overload_coeff) : chassis.normal_step_energy_drain
	chassis.occupant_message("<span class='danger'>You [chassis.leg_overload_mode ? "enable" : "disable"] leg actuators overload.</span>")
```

Reply:
idk why you were setting chassis.leg_overload_mode in that way it did nothing

Reply:
sorry vestigial code